### DISCUSSION TITLE
Fastfile code language and how to optional func and variables

### DISCUSSION THREAD

**Discussion by yildirimatcioglu** [fastlane/fastlane#17852] [Q&A] [answered]
Hello folks I want to ask you to what is the code language of "fastfile" and how can I learn the syntax.
The main question about the optional values and passing optional params to defined functions.
How can I set an optional value to a defined function.?
How can I make an optional checks like swift `optionalValue ?? "something"` as an example.

**Comment by rogerluan** [comment, marked as answer]
Hey @yildirimatcioglu 👋 

The `Fastfile` is written in Ruby. However, if you're more comfortable with Swift, _fastlane_ supports configuration files written in Swift via `Fastfile.swift`. Check you [Getting Started with Fastlane.swift (beta)](https://docs.fastlane.tools/getting-started/ios/fastlane-swift/)

In Ruby I think you can pass fallback values such as `optionalValue ?? "something"` as well as `optionalValue || "something"`. Not sure if the 1st option works in all versions of Ruby, but the 2nd definitely does.
